### PR TITLE
Add implementation of Exit command

### LINKED ISSUE #248: Add support for Exit command
#### Description:

Add implementation of Exit command added in https://github.com/BHoM/BHoM_Adapter/issues/302

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### NOTE: Depends on 

https://github.com/BHoM/BHoM_Adapter/pull/303

 ### Issues addressed by this PR

 Closes #248

 

Adding support for the Exit command

 ### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EjpTgU6jo_tJr30rbJs0a_gBnZnhI1rYcv3aQ-Yc3A6E2g?e=fenJ0J

 ### Changelog

 ### Additional comments

### CODE REVIEW

File: GSA_Adapter/AdapterActions/Execute.cs [hunk 1]
```diff
@@ -134,6 +143,34 @@ public bool RunCommand(Open command)
...
+                    return false;
+                }
+            }
+
+            using (System.Diagnostics.Process gsaProcess = System.Diagnostics.Process.GetProcessById(m_gsaCom.ProcessID()))
+            { 
+                string name = gsaProcess.ProcessName;
+                if (!name.ToUpper().Contains("GSA"))
+                {
+                    Engine.Reflection.Compute.RecordError("Could not find the running GSA process");
```
Comment (peterjamesnugent) [inline GSA_Adapter/AdapterActions/Execute.cs:162]:
```suggestion
                    Engine.Reflection.Compute.RecordError("Could not find the running GSA process.");
```

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Missing punctuation but otherwise code looks good - unable to test with GSA.

**Review by peterjamesnugent — APPROVED** [review]

**Review by alelom — DISMISSED** [review]
LGTM. Tested and behaving as it should. 

Suggested a couple of small changes that I believe good practices, not essential.  
If accepted I will quickly re-approve.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

**Review by alelom — DISMISSED** [review]
Re-approving

**Review by FraserGreenroyd — APPROVED** [review]
@alelom previously approved this, @IsakNaslundBh has only added a `null-check` in his final commit. Owing to @alelom being on leave, I am approving this to unblock @IsakNaslundBh based on the final commit, and leaving merge decisions to @IsakNaslundBh on the basis that I have not actively tested this, only reviewed the final commit.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check core

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**